molecule_libvirt/playbooks/create.yml

```diff
@@ -0,0 +1,259 @@
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/user-data"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create meta-data from template
+      template:
+        src: meta-data.j2
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/meta-data"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create cloud-init iso file
+      command:
+        "mkisofs -o
+        {{ libvirt_default_image_path }}/{{ item.name }}-cloud-init.iso
+        -V cidata -J -r user-data meta-data"
+      args:
+        chdir: "{{ temp_folder }}/{{ item.name }}-molecule"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Remove tempfolfer
+      file:
+        path: "{{ temp_folder }}/{{ item.name }}-molecule"
+        state: absent
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create an image shell with the desired size
+      command:
+        "qemu-img create -f qcow2 -o preallocation=metadata
+        {{ item.name }}-temp.qcow2
+        {{ disk_size | default(default_disk_size ) }}"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      environment:
+        LIBGUESTFS_BACKEND: direct
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Expand original image on the new sized volume
+      command:
+        "virt-resize --quiet --expand /dev/sda1 {{ item.name }}.qcow2
+        {{ item.name }}-temp.qcow2"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Move temporary file to main image
+      command:
+        "mv {{ item.name }}-temp.qcow2 {{ item.name }}.qcow2"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host if item.libvirt_host is defined else 'localhost' }}"
+
+    - name: Define vm from xml file
+      virt:
+        command: define
+        xml: "{{ lookup('template', 'vm/vm.xml.j2') }}"
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
`with_items` has been changed for `loop`.